src/geometric_utils.ts

```diff
@@ -126,17 +214,178 @@ export class GeometricUtils {
         }
     }
 
-    public static get_nearest_point_on_bounding_box(ref_x: number, ref_y: number, spatial_payload: ULabelSpatialPayload2D, dstmax: number = Infinity): object {
-        var ret = {
+    // Return a list of polygons that define all intersections between a list of polygons
+    public static get_polygon_intersections(polygons: ULabelSpatialPayload2D[]): ULabelSpatialPayload2D[] {
+        let ret: ULabelSpatialPayload2D[] = [];
+        for (let i: number = 0; i < polygons.length; i++) {
+            for (let j: number = i+1; j < polygons.length; j++) {
+                let poly1: ULabelSpatialPayload2D = polygons[i];
+                let poly2: ULabelSpatialPayload2D = polygons[j];
+                // Ensure both polygons are closed
+                if (GeometricUtils.is_polygon_closed(poly1) && GeometricUtils.is_polygon_closed(poly2)) {
+                    let intersection: ULabelSpatialPayload2D = GeometricUtils.get_polygon_intersection_single(poly1, poly2);
+                    if (intersection != null) {       
+                        // Don't add duplicate intersections
+                        let is_duplicate: boolean = false;
+                        for (let k: number = 0; k < ret.length; k++) {
+                           if (GeometricUtils.polygons_are_equal(ret[k], intersection)) {
+                               is_duplicate = true;
+                               break;
```

Comment:
Something like the below snippet would be a way to reduce [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) and improve readability. (Disclaimer: untested code)

```js
// getCombinations(['apple', 'banana', 'cucumber']) => [ [ 'apple', 'banana' ], [ 'apple', 'cucumber' ], [ 'banana', 'cucumber' ] ]
function getCombinations (array) {
  return _.flatMap(array, (value, index, collection) =>
    _.map(_.slice(collection, index + 1), (secondValue) => [value, secondValue])
  )
}

public static get_polygon_intersections(polygons) {
  const closedPolygons = _.filter(polygons, GeometricUtils.is_polygon_closed)

  const combinations = getCombinations(closedPolygons)

  const intersections = _.map(combinations, GeometricUtils.get_polygon_intersection_single)

  const nonNullIntersections = _.compact(intersections)

  const deduplicatedIntersections = _.uniqBy(nonNullIntersections, _.isEqual)
}
```